tools/syz-declextract/syz-declextract.cpp

```diff
@@ -106,31 +109,63 @@ bool beginsWith(const std::string_view &str, const std::string_view begin) {
 
 bool contains(const std::string_view &str, const std::string_view sub) { return str.find(sub) != std::string::npos; }
 
-bool isPadding(const std::string &fieldName) {
-  return contains(fieldName, "pad") || contains(fieldName, "unused") || contains(fieldName, "_reserved");
+const std::string int8Subtype(const std::string &name) { return "int8"; }
+const std::string int16Subtype(const std::string &name) {
+  if (contains(name, "port")) {
+    return "sock_port";
+  }
+  return "int16";
 }
-
-const std::string makeArray(const std::string &type, const size_t len = 0) {
-  if (len == 1) {
-    return type;
+const std::string int32Subtype(const std::string &name) {
+  if (contains(name, "ipv4")) {
+    return "ipv4_addr";
   }
-  if (len) {
-    return "array[" + type + ", " + std::to_string(len) + "]";
+  if (endsWith(name, "_pid") || endsWith(name, "_tid") || name == "pid" || name == "tid") {
+    return "pid";
   }
-  return "array[" + type + "]";
+  if (endsWith(name, "dfd") && !endsWith(name, "oldfd")) {
+    return "fd_dir";
+  }
+  if (endsWith(name, "_uid") || name == "uid") {
+    return "uid";
+  }
+  if (endsWith(name, "_gid") || name == "gid") {
+    return "gid";
+  }
+  if (endsWith(name, "fd") || endsWith(name, "fd_in") || endsWith(name, "fd_out")) {
```

Comment:
I think it's more generic to say `startsWith(name, "fd_")` instead of `endsWith(name, "fd_in") || endsWith(name, "fd_out")`.